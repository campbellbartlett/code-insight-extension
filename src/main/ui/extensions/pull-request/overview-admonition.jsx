import React from 'react';
import PropTypes from 'prop-types';
import Admonition from 'react-admonitions';

export const OverviewAdmonition = ({ userAdmin }) => (
    <div>
        {userAdmin && (
            <Admonition type="tip">
                As an administrator of this repository you can enable merging of this pull request
                before all code insight quality reports have passed.
            </Admonition>
        )}
    </div>
);

OverviewAdmonition.propTypes = {
    userAdmin: PropTypes.bool,
};

OverviewAdmonition.defaultProps = {
    userAdmin: false,
};
