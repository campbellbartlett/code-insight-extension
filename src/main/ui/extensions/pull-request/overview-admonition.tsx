import React from 'react';
import Admonition from 'react-admonitions';

type OverviewAdmonitionProps = {
    userAdmin: boolean;
};

export const OverviewAdmonition: React.FC<OverviewAdmonitionProps> = (
    props: OverviewAdmonitionProps
) => {
    const { userAdmin } = props;
    return (
        <div id="overview-admonition">
            {userAdmin && (
                <Admonition type="tip">
                    As an administrator of this repository you can enable merging of this pull
                    request before all code insight quality reports have passed.
                </Admonition>
            )}
        </div>
    );
};
