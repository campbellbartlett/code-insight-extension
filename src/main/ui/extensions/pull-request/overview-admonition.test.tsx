import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';
import { OverviewAdmonition } from './overview-admonition';

describe('overview admonition renders correctly', () => {
    it('should render correctly', () => {
        const overviewAdmonition = shallow(<OverviewAdmonition userAdmin />);

        expect(shallowToJson(overviewAdmonition)).toMatchSnapshot();
    });

    it('should show correct message for admin', () => {
        const overviewAdmonition = shallow(<OverviewAdmonition userAdmin />);

        expect(overviewAdmonition.find('Admonition').length).toEqual(1);
    });

    it('should not show a message for non-admin', () => {
        const overviewAdmonition = shallow(<OverviewAdmonition userAdmin={false} />);

        expect(overviewAdmonition.find('Admonition').length).toEqual(0);
    });
});
