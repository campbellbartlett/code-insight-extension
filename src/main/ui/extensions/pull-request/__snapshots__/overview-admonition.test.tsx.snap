// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overview admonition renders correctly should render correctly 1`] = `
<div
  id="overview-admonition"
>
  <Admonition
    type="tip"
  >
    As an administrator of this repository you can enable merging of this pull request before all code insight quality reports have passed.
  </Admonition>
</div>
`;
